EIPS/eip-7928.md

```diff
@@ -0,0 +1,199 @@
+
+2. **Storage value inclusion**: Including post-execution values for write operations facilitates state reconstruction during syncing, enabling faster chain catch-up.
+
+3. **Balance diffs**: Including balance diffs ensures correct handling of execution that depends on account balances, crucial for parallel transaction execution.
+
+4. **Nonce diffs**: Providing pre-block nonce values for transaction senders enables validators to verify nonce ordering without sequential execution.
+
+5. **Efficiency**: An analysis of historical blocks (random sample of 1000 blocks between 22,195,599 and 22,236,441) shows that BALs would have had an average size of around 57 KiB with balance diffs adding only 9.6 KiB on average, which is a reasonable overhead given the performance benefits.
+
+6. **Transaction independence**: Analysis of the same block sample revealed that approximately 60-80% of transactions within a block are fully independent from one another, meaning they access disjoint sets of addresses and storage slots.
+
+### Block Size Considerations
+
+Including access lists increases block size, potentially impacting network propagation times. Based on analysis of historical blocks:
+
+- Average BAL size over 1,000 blocks was around 57 KiB (SSZ-encoded)
+- Average balance diff size was approximately 9.6 KiB
+- Worst-case BAL size for consuming the entire block gas limit with storage access operations would be approximately 0.93 MiB
+- Worst-case balance diff size would be around 0.11 MiB
+
+These sizes are manageable and less than the current worst-case block size achievable through calldata.
+
+### Asynchronous Validation
+
+Block execution can proceed with parallel IO and parallel EVM operations, with verification of the access list occurring alongside execution, ensuring correctness without delaying block processing.
+
+## Backwards Compatibility
+
+This proposal requires changes to the block structure that are not backwards compatible and require a hard fork.
+
+## Security Considerations
+
+### Validation Overhead
+
+Nodes MUST verify that the access lists and balance diffs in the block accurately reflect the actual accesses and balance changes during execution. Discrepancies MUST result in block invalidation.
+
+### Block Size
+
+Including comprehensive access lists, balance diffs and nonce values increases block size, potentially impacting network propagation times. However, as noted in the rationale section, the overhead is reasonable given the performance benefits, with typical BALs averaging around 67 KiB in total.
+
+### State Sync Benefits
+
+The inclusion of post-execution values for writes in BALs provides an additional benefit for state syncing. Nodes can use these values to reconstruct state without processing all transactions, verifying correctness by comparing the derived state root to the head block's state root.
```

Comment:
Yes this is an awesome point which I did not realize so far..! Something to think about a bit more! :thinking:

Reply:
move in motivation/spec section but out of security consideration section